File: 0035-search-insert-position/0035-search-insert-position.kt
class Solution {
    fun searchInsert(nums: IntArray, target: Int): Int {
        var lo = 0
        var hi = nums.lastIndex + 1
        var mid = 0
        while (lo < hi) {
            mid = lo + (hi-lo)/2
            if (nums[mid] < target) {
                lo = mid + 1
            } else {
                hi = mid
            }
        }
        return lo
    }
}

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        lo, hi = -1, len(nums)
        while hi-lo > 1:
            mid = (lo + hi) // 2
            if nums[mid] < target:
                lo = mid
            else:
                hi = mid
        return hi